### ISSUE TITLE
Support AWS Cloudwatch for metrics

### ISSUE THREAD

**Issue by Jeffail** [Jeffail/benthos#191]
It would be good to support AWS Cloudwatch for a metrics target, this would help when deploying Benthos as a lambda function. I'm looking for help on this one since it's possibly a large chunk of work and I'm not particularly familiar with cloudwatch.

**Comment by dougnukem** [comment]
Have you considered implementing metrics using OpenCensus(https://opencensus.io)

It supports multiple types of exporters for stats/tracing etc: /exporters/supported-exporters/go/
- Prometheus
- AWS X-Ray

Also supports 3rd party plugins for exporters like:
- https://github.com/cep21/cwopencensusexporter

Although it also sounds like there's another project OpenTelemetry that might also supercede this, but sounds like they plan on bridging to OpenCensus:
- https://github.com/open-telemetry/opentelemetry-go

**Comment by Jeffail** [comment]
Hey @dougnukem, it was a while since I last checked those projects out but I'm definitely interested, but also a little hesitant to invest time into it until the dust has settled from the projects converging. I'm already using opentracing for the jaeger tracing output so it makes sense to migrate both tracing and some of the metrics types to opentelemetry eventually.

On the other hand I'm also quite attached to the idea of implementing a proper cloudwatch exporter for benthos, from glancing through https://github.com/cep21/cwopencensusexporter#limitations I think there's justification for it. Last time I looked into it I concluded there's a fair amount of work involved (much more than the existing metric exporters) so this might just be a pipe dream.
